include/zio/zio_attr.h

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2019 Thomas Burdick <thomas.burdick@gmail.com>
+ * Copyright (c) 2019 Kevin Townsend
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief ZIO attribute definitions.
+ */
+
+#ifndef ZEPHYR_INCLUDE_ZIO_ATTR_H_
+#define ZEPHYR_INCLUDE_ZIO_ATTR_H_
+
+#include <zephyr/types.h>
+#include <device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief ZIO attribute enum and struct definitions.
+ * @defgroup zio_attributes ZIO attribute definitions
```

Comment:
We'll also need to add a .rst file that references this group name to have the zio_attributes documentation appear in our API docs.